addons/mrp/models/stock_move.py

```diff
@@ -22,6 +23,10 @@ def _compute_picking_type_id(self):
             line_to_remove |= line
         return super(StockMoveLine, self - line_to_remove)._compute_picking_type_id()
 
+    def _search_picking_type_id(self, operator, value):
+        res = super()._search_picking_type_id(operator=operator, value=value)
```

Comment:
if the operator are negative like `not ilike` or `!=` it should be and AND expression (if the value is not False). maybe a test to cover these cases ?